GitUI/CommandsDialogs/RevisionFileTreeControl.cs

```diff
@@ -51,26 +51,29 @@ public partial class RevisionFileTreeControl : GitModuleControl
         private readonly Stack<string> _lastSelectedNodes = new();
         private readonly IRevisionFileTreeController _revisionFileTreeController;
         private readonly IFullPathResolver _fullPathResolver;
-        private readonly IFindFilePredicateProvider _findFilePredicateProvider;
+        private readonly IFindFilePredicateProvider _findFilePredicateProvider = new FindFilePredicateProvider();
         private GitRevision? _revision;
         private readonly RememberFileContextMenuController _rememberFileContextMenuController
             = RememberFileContextMenuController.Default;
         private Action? _refreshGitStatus;
+        private RevisionGridControl? _revisionGrid;
 
         public RevisionFileTreeControl()
         {
             InitializeComponent();
             InitializeComplete();
             HotkeysEnabled = true;
             _fullPathResolver = new FullPathResolver(() => Module.WorkingDir);
-            _findFilePredicateProvider = new FindFilePredicateProvider();
             _revisionFileTreeController = new RevisionFileTreeController(() => Module.WorkingDir,
                                                                          new GitRevisionInfoProvider(() => Module),
                                                                          new FileAssociatedIconProvider());
+            BlameControl.HideCommitInfo();
+            blameToolStripMenuItem1.Checked = AppSettings.RevisionFileTreeShowBlame;
         }
 
-        public void Bind(Action refreshGitStatus)
+        public void Bind(RevisionGridControl revisionGrid, Action? refreshGitStatus)
         {
+            _revisionGrid = revisionGrid;
```

Comment:
I very much dislike this - this creates further coupling between various controls, and making it harder to reason about each individual control's functionality, and further complicating testing stories.

Reply:
This is required for BlameControl so the same as in FormFileHistory.
You can see this as a step to remove a lot of code in GE: The follow up #9445 obsoletes FormFileHistory, so this is a step to remove a complete form.